opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/drools/DroolsAlarmContext.java

```diff
@@ -180,19 +184,43 @@ public void handleDeletedAlarm(int alarmId, String reductionKey) {
         }
     }
 
+    private void handleRelatedAlarms(OnmsAlarm situation) {
+        if (!situation.isSituation()) {
+            return;
+        }
+        Map<Integer, AlarmAssociationAndFact> associationFacts = alarmAssociationById.get(situation.getId());
+        if (associationFacts == null) {
```

Comment:
Can be replaced with call to `computeIfAbsent`.

Reply:
Thanks. will change.
I'm still living in Java 1.2....

Reply:
done.